### PR TITLE
add index populate events

### PR DESCRIPTION
**Pull request by oleg-andreyev**
In my project we are using bulk indexing, and according to [doc](http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/indices-update-settings.html#bulk) to increase performant for bulk indexing, we need to have ability call custom actions like: `refresh_interval` and `_optimize?max_num_segments=5`

To achieve that we've added `PRE_INDEX_POPULATE`, `PRE_TYPE_POPULATE`, `POST_TYPE_POPULATE`, `POST_INDEX_POPULATE`, `INDEX_RESET` events.

### CODE REVIEW

**Comment by merk** [comment]
@oleg-andreyev ping

File: EventListener/PopulateListener.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+     * @var Resetter
+     */
+    private $resetter;
+
+    /**
+     * @param Resetter $resetter
+     */
+    public function __construct(Resetter $resetter)
+    {
+
```
Comment (oleg-andreyev) [inline EventListener/PopulateListener.php]: done

File: Command/PopulateCommand.php [hunk 2]
```diff
@@ -141,19 +130,48 @@ private function populateIndex(OutputInterface $output, $index, $reset, $options
...
+
+    /**
+     * @param OutputInterface     $output
+     * @param ProviderInterface[] $providers
+     * @param string              $index
+     * @param string              $type
+     * @param boolean             $reset
+     * @param array               $options
+     */
+    private function populate(OutputInterface $output, $providers, $index, $type, $reset, $options)
```
Comment (merk) [inline Command/PopulateCommand.php]: Since providers is always required we can typehint array here

**Comment by merk** [comment]
It looks like you've removed the RESET event entirely, I think it still has a useful purpose but needs to fire inside the Resetter.

**Comment by oleg-andreyev** [comment]
@merk, I've noticed that `ResetterTest` is disabled, is there a ticket for that, I can address it.

**Comment by merk** [comment]
I dont believe there is at the moment, I would appreciate it if you could look into fixing the tests for that class. Just create a new PR

**Comment by dupuchba** [comment]
any news on this one ?

**Comment by oleg-andreyev** [comment]
ping @merk

**Comment by merk** [comment]
@oleg-andreyev I've done a bit more work to get this mergable with current master - Any chance you could work on a cookbook entry to add to the docs so there is some documentation on how to use these events?

A pr for others to review based off this one: #814

**Comment by oleg-andreyev** [comment]
Okay, I'll create new PR for cookbook

**Pull request closed without merging**